Wikipedia/Code/EditSummaryViewController.swift

```diff
@@ -65,7 +72,7 @@ import UIKit
         delegate?.learnMoreButtonTapped(sender: sender)
     }
 
-    public func textFieldDidChange(textField: UITextField){
+    @objc public func textFieldDidChange(textField: UITextField){
         notifyDelegateOfSummaryChange()
```

Comment:
wouldn't using `textField(_ textField: UITextField, shouldChangeCharactersIn range: NSRange, replacementString string: String)` be enough to detect changes?

Reply:
Ah will check thx!

Reply:
In this case `didChange` seems to be the better place to check that a change happened. If `shouldChange` was notified when the text was changed programmatically (when we tap one of the canned summary description buttons) I'd probably switch it so we didn't have to manually notify the delegate on button tap, but unfortunately neither `shouldChange` nor `didChange` get called when we change the text programmatically.